clippy_lints/src/shadow.rs

```diff
@@ -158,11 +158,22 @@ impl<'tcx> LateLintPass<'tcx> for Shadow {
     }
 }
 
-fn is_shadow(cx: &LateContext<'_>, owner: LocalDefId, first: ItemLocalId, second: ItemLocalId) -> bool {
+type Result<T> = std::result::Result<T, LifetimeError>;
+
+#[derive(Debug, Clone)]
+struct LifetimeError;
+
+impl std::fmt::Display for LifetimeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Invalid lifetime")
+    }
+}
+
+fn is_shadow(cx: &LateContext<'_>, owner: LocalDefId, first: ItemLocalId, second: ItemLocalId) -> Result<bool> {
```

Comment:
The method `is_***` should return `bool`. And now, `var_scope()` return `Option<T>`, so it would be better to use `if let` like below:
```rs
fn is_shadow(cx: &LateContext<'_>, owner: LocalDefId, first: ItemLocalId, second: ItemLocalId) -> bool {
    let scope_tree = cx.tcx.region_scope_tree(owner.to_def_id());
    if let Some(first_scope) = scope_tree.var_scope(first) {
        if let Some(second_scope) = scope_tree.var_scope(second) {
            return scope_tree.is_subscope_of(second_scope, first_scope);
        }
    }

    false
}
```